## Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'])

In [4]:
# Set the date column as the dataframe index + drop date column
df.set_index('date', inplace=True, drop=True)

### Compare June and December data across all years

In [5]:
from scipy import stats

In [6]:
# Filter data for desired months
june_dec = df.loc[(df.index.get_level_values('date').strftime('%m') == '06') | \
    (df.index.get_level_values('date').strftime('%m') == '12')]

june_dec.head()

station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77

In [7]:
# Identify the average temperature for June
june_dec.loc[(june_dec.index.get_level_values('date').strftime('%m') == '06'), 'tobs'].mean()

74.94411764705882

In [8]:
# Identify the average temperature for December
june_dec.loc[(june_dec.index.get_level_values('date').strftime('%m') == '12'), 'tobs'].mean()

71.04152933421226

In [9]:
# Create collections of temperature data
june = june_dec.loc[(june_dec.index.get_level_values('date').strftime('%m') == '06'), 'tobs']
dec = june_dec.loc[(june_dec.index.get_level_values('date').strftime('%m') == '12'), 'tobs']

In [10]:
# Run unpaired t-test
stats.ttest_ind(june, dec, equal_var=False)

Ttest_indResult(statistic=31.355036920962423, pvalue=4.193529835915755e-187)

### Analysis
Because the June and December temperature data classify as two distinct populations (rather than being compared to a standard value), an independent (unpaired) t-test is more appropriate. In general, the larger the t-test statistic, the more difference exists between the two sets being compared. Comparing temperatures from June and December produced a statistic of approximately 31.36, suggesting that the groups have a relatively large difference. Additionally, the p-value is less than 0.05 ($p \approx 4.19*10^{-187}$) rejects the null hypothesis that there is no significant difference between the two months' temperatures. 
